containers/ReferenceValue.js

```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import { connect } from 'react-redux'
+import { changeReferenceValue } from '../actions'
+
+let ReferenceValue = ({ entityId, refId, referenceValue, dispatch }) => (
+  <div>
+    <input type="text" 
+      placeholder="reference value"
+      value={referenceValue}
+      onChange={e => dispatch(changeReferenceValue(entityId, refId, e.target.value))}/>
```

Comment:
I suggest not to put it in container. (same for Entity, EntityName, ReferenceValue, SynonymList)
Put them in component and pass the function from parent.
Ref: Todo and TodoList in todos, redux examples